pkg/registry/core/pod/strategy.go

```diff
@@ -360,11 +360,8 @@ func (p podResizeStrategy) GetResetFieldsFilter() map[fieldpath.APIVersion]field
 // dropPodUpdates drops any changes in the pod.
 func dropPodUpdates(newPod, oldPod *api.Pod) *api.Pod {
 	pod := oldPod.DeepCopy()
-	pod.Name = newPod.Name
-	pod.Namespace = newPod.Namespace
 	pod.ResourceVersion = newPod.ResourceVersion
```

Comment:
managed fields also seems important to preserve

the _normal_ approach status strategies take is to stomp spec unconditionally and call ResetObjectMetaForStatus to selectively stomp metadata fields

this and ephemeral containers strategies are doing pretty unusualy things projecting spec fields into subresources, so I'm not sure there's a lot of examples to follow

Reply:
> the normal approach status strategies take is to stomp spec unconditionally and call ResetObjectMetaForStatus to selectively stomp metadata fields

Looked into this a bit more and I think we can still do this with resize and ephemeral containers strategies too. Currently `dropPodUpdates` seems to be doing exactly that (or at least trying to) with the strategies then resetting the spec fields as needed, so we can keep doing that but just use the common ` ResetObjectMetaForStatus` instead of having the custom `dropPodUpdates`

I updated the PR so you can see what that looks like, let me know what you think